Add more efficient workaround for bug #27504

amahmudTT opened the pull request:
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/27504?reload=1)

### Problem description
This is a more efficient workaround for the issue mentioned. The issue is more than what is described, that is packer reconfiguration is not working properly. We need to call an uninit function  corresponding to llk_unpack_tilizeA_B_init. If we do that it works for all cases except when output format is bfp4 and the tensor is of particular size. I am yet to figure out why. My assumption is that a reconfig_data_format call is needed in that case, but even that function may require revamping as it is not working. So the temporary fix is use the fixed pack_reconfig_data format along with a fuill blown llk_unpack_A_hw_configure_disaggregated


### What's changed
This PR just integrates the packer reconfiguration fix and adds a more efficient workaround for the unpacker config issues. 


### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

amahmudTT committed:
Add more efficient workaround for bug #27504

skrsticTT commented on ttnn/cpp/ttnn/operations/pool/generic/device/kernels/compute/compute_pool_2d.cpp:
```diff
@@ -292,8 +292,10 @@ void MAIN {
                         // Workaround until #27504 is not closed
                         tensix_sync();
-                        unary_op_init_common(pre_tilize_cb_id, out_cb_id);
+                        UNPACK((llk_unpack_A_hw_configure_disaggregated<DST_ACCUM_MODE, StochRndType::None, false>(
```
remove unary header include if not used anymore

skrsticTT commented:
can you please add some short description what is changed? also can you rename PR name, and attach running/passing pipelines?

amahmudTT commented:
> > I added it as a comment in the code. Please check.
> 
> I thought it might be clearer if the PR title and description were a bit more descriptive
> 
> > Running the tests.
> 
> Okay please post passing pipelines in the PR description as well

Added to description.

amahmudTT commented:
Finally a passing nightly test. https://github.com/tenstorrent/tt-metal/actions/runs/18669155643

Trying to merge
